ens-de-lyon-centre-d-ingenierie-documentaire.csl

```diff
@@ -338,18 +376,47 @@
       </else>
     </choose>
   </macro>
+  <!-- MACRO PAGES PAR TYPE DE DOCUMENT (ndbp et biblio)-->
+  <macro name="pages">
+    <choose>
+      <!-- Pour les textes qui sont déjà une partie d'ouvrage -->
+      <if type="article-journal article-newspaper article-magazine chapter paper-conference entry-dictionary entry-encyclopedia" match="any">
+        <choose>
+          <if variable="locator" match="any">
+            <text variable="locator" prefix="p.&#160;"/>
+          </if>
+          <else-if variable="locator" match="none">
+            <label variable="page" form="short" suffix="&#160;"/>
+            <text variable="page"/>
+          </else-if>
+        </choose>
+      </if>
+      <!-- Pour les autres cas -->
+      <else>
+        <choose>
```

Comment:
this choose/if isn't necessary (if there's no locator, CSL won't print anything anyway)